cdn/sign_url.rb

```diff
@@ -0,0 +1,64 @@
+# Copyright 2018 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def signed_url url:, key_name:, key:, expiration:
+  # [START signed_url]
+  # url        = "URL of the endpoint served by Cloud CDN"
+  # key_name   = "Name of the signing key added to the Google Cloud Storage bucket or service"
+  # key        = "Signing key as urlsafe base64 encoded string"
+  # expiration = "Expiration date and time for the signed URL"
+
+  require "base64"
+  require "openssl"
+  require "time"
+
+  # Decode the URL safe base64 encode key
+  decoded_key = Base64.urlsafe_decode64 key
+
+  # Determine UTC time for given date and time
+  expiration = Time.parse(expiration).utc.to_i
```

Comment:
are you now accepting `expiration` as string? I thought you were going to take `Time` directly.

Do you think users of this method will compute now()+some_time and turn it into a string to sign the URL?

Reply:
I'm not sure I understand. The time offset needs to be given so as an example they can do Time.parse(expiration) if they have a specific date and time and then follow-up with an update this sample to just use now() + some_time.

I can update the value of expiration to be the number of seconds + now().

Reply:
I guess we need to clarify what you mean by `expiration` here.  Is it `expirationDate` or `expiresIn`?

If it's `expirationDate` (which is what other samples do), then it should accept a type that contains the date/time information.

Therefore we want to end up with an example that's like this, in Python:

```py
tomorrow = datetime.now(), timedelta(days=1)
signed_url = sign_url(orig_url, tomorrow, ...)
```

In this case `sign_url` is accepting the `datetime` type; not the string (`str`) representation of the date. Is it clearer now?

Reply:
Followed-up offline, for clarity.